### PR TITLE
feat: use built-in input parsing & add `cargo.lock`

### LINKED ISSUE #103: Bump Serenity to v0.10.10
Always good to bump Serenity. Also unlocks #97.

### PR DESCRIPTION
**Pull request by afonsojramos**
Closes #103, #97 and #110.

image

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ branch = "current"
 features = ["builtin-queue", "yt-dlp"]
 
 [dependencies.serenity]
-version = "0.10"
+version = "0.10.10"
```
Comment (joao-conde) [inline Cargo.toml:28]: Weird, `0.10` should fetch the latest patch version, right? Which would make it `0.10.10`? Have you tried with just `0.10` but making `cargo clean` and then `cargo run` ?

Comment (aquelemiguel) [reply Cargo.toml:28]: Can confirm, just tested and all is working when when keeping `0.10`.

Comment (afonsojramos) [reply Cargo.toml:28]: Missed the `cargo clean` rolling it back!

Comment (afonsojramos) [reply Cargo.toml:28]: For some reason, even after a `cargo clean` it uses `v0.10.9`. Did you try `cargo clean` in this branch @aquelemiguel?

**Review by joao-conde — APPROVED** [review]

**Pull request merged**